### DISCUSSION TITLE
Give the Godot shading language a proper name

### DISCUSSION THREAD

**Discussion by DeeJayLSP** [godotengine/godot-proposals#4334] [General Discussions]
Currently, the Godot shading language it's known, according to the docs, as "Godot shading language", "Godot's shading language", "a shading language based on the popular GLSL" or "a shading language similar to GLSL ES 3.0". 

I propose the adoption of a formal name to represent the language, such as "GDSL" or "GodotSL"

**Comment by Calinou** [comment]
I think "Godot shader language" is as proper as it can get :slightly_smiling_face: 

"GDSL" has a higher potential for confusion with GLSL, especially among beginners.

**Reply by DeeJayLSP** [reply]
Just like HTML stands for Hypertext Markup Language and CSS stands for Cascading Style Sheets, we could use something that stands for Godot Shader Language. Both GLSL and HLSL are short forms of shading language names.

I think it would be way better to document if there's a short form that points out to it. As a default, most Shader languages end with SL, so `<something that points out to Godot>SL` would be easily understandable. Of course, it's better that the short form doesn't have a potential for confusion.

Currently, "GodotSL" is my best bet, even if it looks a bit generic.

**Comment by AlfishSoftware** [comment]
In [my VSCode plugin](https://github.com/AlfishSoftware/godot-files-vscode) I'm referring to it as both "GDShader" (the filename extension) and "Godot Shader".
As for the INI-like format Godot uses for .tscn, .tres, .import, etc, I'm calling it "GDAsset" when referring to the common grammar/syntax for them.
Since we have "GDScript", I think it would make sense for those names to be made official, in the same pattern.

**Comment by Kluskey** [comment]
+1 to this. We're working on some university coursework and it'd be great to have a specific shader name instead of "Godot shader language" or something informal "Kind of GLSL". GDShader sounds cool and fits in well with the naming convention with GDScript

**Reply by tetrapod00** [reply]
I've been calling it GDShader whenever I need to specify it, for example when making a distinction between actual GLSL.

**Comment by tetrapod00** [comment]
Made a formal proposal for this https://github.com/godotengine/godot-proposals/issues/10784

**Comment by azur-wolve** [comment]
GDShader → `.gds` (but problem: same extension as GDScript)

GDShaderscript → `.gdss`
